spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/utils/StandardMeshUtils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *     Copyright (C) 2016 to the original authors.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *             http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.istio.utils;
+
+import me.snowdrop.istio.client.IstioClient;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+
+
+public class StandardMeshUtils implements MeshUtils {
+
+	private static final Log LOG = LogFactory.getLog(StandardMeshUtils.class);
+
+	private final IstioClient client;
+
+	public StandardMeshUtils(IstioClient client) {
+		if (client == null) {
+			throw new IllegalArgumentException("Must provide an instance of IstioClient");
+		}
+		this.client = client;
+	}
+
+
+	@Override
+	public Boolean isIstioEnabled() {
+		return checkIstioServices();
+	}
+
+	private synchronized boolean checkIstioServices() {
+		try {
+			//Check if Istio proxy is installed.
+			// We can improve this initial detection if better methods are found.
+			WebClient client = WebClient.create("http://localhost:15090/");
+			ClientResponse clientResponse = client.get().uri("stats/prometheus").exchange().block();
```

Comment:
I think the port and path should be configurable. I can envision some Istio installation were these defaults would not apply.

Furthermore, is there way to return immediately once the first byte of an  HTTP 200 is returned?

Reply:
@geoand yeah.. I was checking on how to get just the http code.. but I couldn't find anything for that (on my quick search)

Reply:
@spencergibb any ideas?

Reply:
I will make the URL configurable.. but the problem is that we are relaying on this to be local, so I wouldn't recommend to change this default values.

Reply:
The base for the URL should never change. However the port and the proxy path to be used should be configurable so we can cover all cases

Reply:
`WebClient` isn't always available either since it is in webflux.

Reply:
I will switch to rest tplate here

Reply:
@spencergibbs switching to a bootstrap config right now

Reply:
Also I am not sure that `15090` is the proper default. I mean it is for openshift, but on the microk8s cluster I created the port was `15000` (which also seems to be the default according to istio [docs](https://istio.io/docs/reference/commands/pilot-agent/))